libraries/type-length-value/README.md

```diff
@@ -0,0 +1,104 @@
+# Type-Length-Value
+
+Library with utilities for working with Type-Length-Value structures.
+
+## Example usage
+
+This simple examples defines a zero-copy type with its discriminator.
+
+```rust
+use {
+    bytemuck::{Pod, Zeroable},
+    spl_type_length_value::{discriminator::TlvType, state::{TlvState, TlvStateMut}},
+}
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+struct MyValue {
+    data: [u8; 32],
+}
+impl TlvType for MyValue {
+    const TYPE: Discriminator = Discriminator::new([1; Discriminator::LENGTH]);
+}
+let account_size = TlvState::get_base_len() + std::mem::size_of::<MyValue>();
+
+// Buffer likely comes from a Solana `solana_program::account_info::AccountInfo`,
+// but this example just uses a vector.
+let mut buffer = vec![0; account_size];
+let mut state = TlvStateMut::unpack(&mut buffer).unwrap();
+
+// init and write default value
+let value = state.init_value::<MyValue>().unwrap();
+```
```

Comment:
Can you give an example where multiple different extensions or `TlvType`s are serialized & deserialized from the same slab?

These examples only show 1 type being serialized / deserialized, which doesn't show off the usefulness of TLV lib :)